test/GnosisSafeSameAddressMultisig.js

```diff
@@ -0,0 +1,105 @@
+/*global describe, context, beforeEach, it*/
+
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
+
+describe("GnosisSafeSameAddressMultisig", function () {
+    let gnosisSafe;
+    let gnosisSafeProxyFactory;
+    let gnosisSafeSameAddressMultisig;
+    let signers;
+    let initialOwner;
+    let newOwnerAddresses;
+    const initialThreshold = 1;
+    const newThreshold = 3;
+    const AddressZero = "0x" + "0".repeat(40);
+    const maxUint256 = "115792089237316195423570985008687907853269984665640564039457584007913129639935";
+
+    beforeEach(async function () {
+        const GnosisSafe = await ethers.getContractFactory("GnosisSafe");
+        gnosisSafe = await GnosisSafe.deploy();
+        await gnosisSafe.deployed();
+
+        const GnosisSafeProxyFactory = await ethers.getContractFactory("GnosisSafeProxyFactory");
+        gnosisSafeProxyFactory = await GnosisSafeProxyFactory.deploy();
+        await gnosisSafeProxyFactory.deployed();
+
+        const GnosisSafeSameAddressMultisig = await ethers.getContractFactory("GnosisSafeSameAddressMultisig");
+        gnosisSafeSameAddressMultisig = await GnosisSafeSameAddressMultisig.deploy(gnosisSafe.address);
+        await gnosisSafeSameAddressMultisig.deployed();
+
+        signers = await ethers.getSigners();
+        initialOwner = signers[1];
+        newOwnerAddresses = [signers[1].address, signers[2].address, signers[3].address, signers[4].address];
+    });
+
+    context("Verifying multisigs", async function () {
+        it("Should fail when passing the non-zero multisig data with the incorrect number of bytes", async function () {
+            await expect(
+                gnosisSafeSameAddressMultisig.create(newOwnerAddresses, initialThreshold, "0x55")
+            ).to.be.revertedWith("IncorrectDataLength");
+
+            const data = AddressZero + "55";
+            await expect(
+                gnosisSafeSameAddressMultisig.create(newOwnerAddresses, initialThreshold, data)
+            ).to.be.revertedWith("IncorrectDataLength");
+        });
+
+        it("Create a multisig and change its owners and threshold", async function () {
```

Comment:
Verifying the create() function that in reality does the verification of a new multisig setup